### PR TITLE
fix: List focus via keyboard

### PR DESCRIPTION
**Pull request by umeshiscreative**
#### What does this PR do?
fix the bug when the List focus from the keyboard.

#### Where should the reviewer start?
component/List/List.js

#### What testing has been done on this PR?
Updated the snapshot for this component

#### How should this be manually tested?
I used the storybook to test this component.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
The list focuses on accepting only from the keyboard.

#### What are the relevant issues?
#6589

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
No

### CODE REVIEW

**Comment by umeshiscreative** [comment]
@jcfilben Can you please review this PR?

File: src/js/components/List/List.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ const StyledList = styled.ul`
     props.itemFocus &&
```
Comment (taysea) [inline src/js/components/List/List.js:52]: This is looking great, although it looks like this block from props.itemFocus is causing the focus indicator to still appear when interacting with mouse. I don't believe this block is necessary anymore given the latest focus/unfocus styles. Can you remove this block entirely? I checked on Chrome, Safari, and Firefox and it behaved as expected for mouse + keyboard navigation:

comment + code to be removed:
```
 // during the interim state when a user is holding down a click,
  // the individual list item has focus in the DOM until the click
  // completes and focus is placed back on the list container.
  // for visual consistency, we want to keep the focus indicator on the
  // list container the whole time.
  ${(props) =>
    props.itemFocus &&
    focusStyle({ forceOutline: true, skipSvgChildren: true })}}
```

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by taysea — APPROVED** [review]
Looks good! Thank you

**Pull request merged**